### PR TITLE
Fix `x` not to quit after `x` prints `settings.json`

### PR DESCRIPTION
**Pull request by long-long-float**
I fixed the `x` not to quit after the `x` prints `.vscode/settings.json`.

The command `x setup` ask as following:

```
x.py can automatically install the recommended `.vscode/settings.json` file for rustc development
Would you like to create/update `settings.json`, or only print suggested settings?: [y/p/N]
```

When user types `p`, the `x` prints the contents and quit the program.
It is a hassle to start the command again, so I fixed the `x` to ask what to do again.

### CODE REVIEW

File: src/bootstrap/src/core/build_steps/setup.rs [hunk 1]
```diff
@@ -549,12 +549,18 @@ impl Step for Vscode {
         if config.dry_run() {
             return;
         }
-        t!(create_vscode_settings_maybe(&config));
+        loop {
+            let created = t!(create_vscode_settings_maybe(&config));
+            if created {
+                break;
+            }
+        }
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/setup.rs:557]:
```suggestion
        while !t!(create_vscode_settings_maybe(&config)) {}
```

**Review by onur-ozkan — COMMENTED** [review]
https://github.com/rust-lang/rust/blob/0a83e43f282353398e4b22466cf1f4e3cc8c1682/src/bootstrap/src/core/build_steps/setup.rs#L590

This message needs to be updated too

**Comment by onur-ozkan** [comment]
> This message needs to be updated too

Something like "Would you like to create/update `settings.json`? (Press 'p' to preview values): [y/N]" would be enough.

**Comment by long-long-float** [comment]
@onur-ozkan Thank you for your review. I applied your suggestions.

**Review by onur-ozkan — APPROVED** [review]
LGTM.

Could you please squash commits into single one?

**Comment by long-long-float** [comment]
@onur-ozkan I did it.

**Pull request merged**